src/state_machine.rs

```diff
@@ -331,6 +331,10 @@ pub enum Transition {
     IntoElder {
         old_pfx: Prefix<XorName>,
     },
+    // `Elder` state transition to `Adult` as a result of demotion
+    Demote {
```

Comment:
Should we rename the `IntoElder` transition to `Promote`, for symmetry?

Reply:
Could do, probably would leave it out of this PR though?

Reply:
OK.